src/main/scala/com/github/gvolpe/fs2/Resources.scala

```diff
@@ -44,31 +40,31 @@ object ResourcesApp extends Resources[IO]
   *
   * Finally, R3 was done showing an availability of one once again.
   * */
-class Resources[F[_]: Effect] extends StreamApp[F] {
+object ResourcesApp extends IOApp {
 
-  override def stream(args: List[String], requestShutdown: F[Unit]): fs2.Stream[F, ExitCode] =
-    Scheduler(corePoolSize = 4).flatMap { implicit scheduler =>
+  def stream[F[_]: ConcurrentEffect: Timer]: fs2.Stream[F, Unit] =
       for {
-        s   <- Stream.eval(async.semaphore[F](1))
+        s   <- Stream.eval(Semaphore[F](1))
         r1  = new PreciousResource[F]("R1", s)
         r2  = new PreciousResource[F]("R2", s)
         r3  = new PreciousResource[F]("R3", s)
-        ec  <- Stream(r1.use, r2.use, r3.use).join(3).drain ++ Stream.emit(ExitCode.Success)
-      } yield ec
-    }
+        _  <- Stream(r1.use, r2.use, r3.use).parJoin(3)
+      } yield ()
+
+  override def run(args: List[String]): IO[ExitCode] =
+    stream[IO].compile.drain.as(ExitCode.Success)
 
 }
 
-class PreciousResource[F[_]: Effect](name: String, s: Semaphore[F])
-                                    (implicit S: Scheduler) {
+class PreciousResource[F[_]: Effect: Timer](name: String, s: Semaphore[F]) {
```

Comment:
Maybe over-constrained?

Reply:
I rewrote ResourcesApp and removed the streams